Compilation error with psa_crypto.c

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure that nothing follows the Issue request type section, all 
   issue details are within the Description section and no changes are made to the 
   template format (as detailed below).

   *************************************************************************************

-->

### Description

I'm getting following compilation error when compiling https://github.com/ARMmbed/mbed-mesh-test-application with custom mbedtls configuration (mbed_app_thread_thci_lite_ATMEL.json):

```
[Error] psa_crypto.c@3151,47: 'union <anonymous>' has no member named 'gcm'; did you mean 'ccm'?
[Error] psa_crypto.c@3265,55: 'union <anonymous>' has no member named 'gcm'; did you mean 'ccm'?
[Error] psa_crypto.c@3351,54: 'union <anonymous>' has no member named 'gcm'; did you mean 'ccm'?
[Warning] psa_crypto.c@1152,33: 'mbedtls_md_info_from_psa' defined but not used [-Wunused-function]
[ERROR] ./mbed-os/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c: In function 'psa_aead_abort':
./mbed-os/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c:3151:47: error: 'union <anonymous>' has no member named 'gcm'; did you mean 'ccm'?
             mbedtls_gcm_free( &operation->ctx.gcm );
                                               ^~~
                                               ccm
./mbed-os/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c: In function 'psa_aead_encrypt':
./mbed-os/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c:3265:55: error: 'union <anonymous>' has no member named 'gcm'; did you mean 'ccm'?
             mbedtls_gcm_crypt_and_tag( &operation.ctx.gcm,
                                                       ^~~
                                                       ccm
./mbed-os/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c: In function 'psa_aead_decrypt':
./mbed-os/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c:3351:54: error: 'union <anonymous>' has no member named 'gcm'; did you mean 'ccm'?
             mbedtls_gcm_auth_decrypt( &operation.ctx.gcm,
                                                      ^~~
                                                      ccm
At top level:
./mbed-os/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c:1152:33: warning: 'mbedtls_md_info_from_psa' defined but not used [-Wunused-function]
 static const mbedtls_md_info_t *mbedtls_md_info_from_psa( psa_algorithm_t alg )
                                 ^~~~~~~~~~~~~~~~~~~~~~~~

```
It looks like some variables in structure `aead_operation_t` are flagged but flags are not always used when accessing the flagged variable.

In addition, once `operation->ctx.gcm` is flagged with MBEDTLS_CCM_C in psa_crypto.c#3151

Used mbed-os: c0108b1b48514a822856eaa8c256940f4cb70c7b


<!--
    Required
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html
    Things to consider sharing:
    - What target does this relate to?
    - What toolchain (name + version) are you using?
    - What tools (name + version - is it mbed-cli, online compiler or IDE) are you using?
    - What is the SHA of Mbed OS (git log -n1 --oneline)?
    - Steps to reproduce. (Did you publish code or a test case that exhibits the problem?)
-->



### Issue request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the issue otherwise.)
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Question
    [ ] Enhancement
    [x] Bug

--- a/features/mbedtls/VERSION.txt
+++ b/features/mbedtls/VERSION.txt
@@ -1,2 +1,2 @@
-mbedtls-2.15.0
-mbedcrypto-0.1.0b
+mbedtls-2.15.1
+mbedcrypto-0.1.0b2

--- a/features/mbedtls/importer/Makefile
+++ b/features/mbedtls/importer/Makefile
@@ -27,7 +27,7 @@
 #
 
 # Set the mbed TLS release to import (this can/should be edited before import)
-MBED_TLS_RELEASE ?= mbedtls-2.15.0
+MBED_TLS_RELEASE ?= mbedtls-2.15.1
 
 # Translate between mbed TLS namespace and mbed namespace
 TARGET_PREFIX:=../

--- a/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c
+++ b/features/mbedtls/mbed-crypto/platform/TARGET_PSA/COMPONENT_PSA_SRV_IMPL/psa_crypto.c
@@ -3146,7 +3146,7 @@ static void psa_aead_abort( aead_operation_t *operation )
             mbedtls_ccm_free( &operation->ctx.ccm );
             break;
 #endif /* MBEDTLS_CCM_C */
-#if defined(MBEDTLS_CCM_C)
+#if defined(MBEDTLS_GCM_C)
         case PSA_ALG_GCM:
             mbedtls_gcm_free( &operation->ctx.gcm );
             break;
@@ -3259,6 +3259,7 @@ psa_status_t psa_aead_encrypt( psa_key_slot_t key,
     }
     tag = ciphertext + plaintext_length;
 
+#if defined(MBEDTLS_GCM_C)
     if( operation.core_alg == PSA_ALG_GCM )
     {
         status = mbedtls_to_psa_error(
@@ -3270,7 +3271,10 @@ psa_status_t psa_aead_encrypt( psa_key_slot_t key,
                                        plaintext, ciphertext,
                                        operation.tag_length, tag ) );
     }
-    else if( operation.core_alg == PSA_ALG_CCM )
+    else
+#endif /* MBEDTLS_GCM_C */
+#if defined(MBEDTLS_CCM_C)
+    if( operation.core_alg == PSA_ALG_CCM )
     {
         status = mbedtls_to_psa_error(
             mbedtls_ccm_encrypt_and_tag( &operation.ctx.ccm,
@@ -3282,6 +3286,7 @@ psa_status_t psa_aead_encrypt( psa_key_slot_t key,
                                          tag, operation.tag_length ) );
     }
     else
+#endif /* MBEDTLS_CCM_C */
     {
         return( PSA_ERROR_NOT_SUPPORTED );
     }
@@ -3339,6 +3344,7 @@ psa_status_t psa_aead_decrypt( psa_key_slot_t key,
     if( status != PSA_SUCCESS )
         return( status );
 
+#if defined(MBEDTLS_GCM_C)
     if( operation.core_alg == PSA_ALG_GCM )
     {
         status = psa_aead_unpadded_locate_tag( operation.tag_length,
@@ -3356,7 +3362,10 @@ psa_status_t psa_aead_decrypt( psa_key_slot_t key,
                                       tag, operation.tag_length,
                                       ciphertext, plaintext ) );
     }
-    else if( operation.core_alg == PSA_ALG_CCM )
+    else
+#endif /* MBEDTLS_GCM_C */
+#if defined(MBEDTLS_CCM_C)
+    if( operation.core_alg == PSA_ALG_CCM )
     {
         status = psa_aead_unpadded_locate_tag( operation.tag_length,
                                                ciphertext, ciphertext_length,
@@ -3374,6 +3383,7 @@ psa_status_t psa_aead_decrypt( psa_key_slot_t key,
                                       tag, operation.tag_length ) );
     }
     else
+#endif /* MBEDTLS_CCM_C */
     {
         return( PSA_ERROR_NOT_SUPPORTED );
     }

